src/calendar/calendar-year.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from "react";
+import dayjs, { Dayjs } from "dayjs";
+import { CalendarYearProps, VariantYear } from "./types";
+import { YearCell, YearPickerContainer } from "./calendar-year.style";
+import { CalendarHelper } from "../util/calendar-helper";
+
+export const CalendarYear = ({
+    calendarDate,
+    showView,
+    selectedStartDate,
+    onSelect,
+    onDecadeChange,
+}: CalendarYearProps) => {
+    const [yearDate, setYearDate] = useState<Dayjs[]>([]);
+
+    useEffect(() => {
+        if (showView === "Year") {
+            generateDecadeOfYears();
+        }
+    }, [showView, calendarDate]);
+
+    const generateYearStatus = (date: Dayjs) => {
+        const grayYearIndex = [0, 11];
+
+        const isDecadeYear = grayYearIndex.includes(yearDate.indexOf(date));
+        const fullDate = date.format("YYYY-MM-DD");
+        const year = fullDate.split("-")[0];
```

Comment:
this can be `date.year()` right?

Reply:
yes.

the FullDate (YYYY-MM-DD) needed also for below code.

isSame("YYYY-MM-DD") ----> format

```
   const variant: VariantYear = isOtherDecade
            ? "other-decade"
            : dayjs(selectedStartDate).isSame(fullDate, "year")
            ? "selected-year"
            : dayjs().isSame(fullDate, "year")
            ? "current-year"
            : "default";

```

Reply:
Updated to this

```
const fullDate = date.format("YYYY-MM-DD");
const year = date.year();
```